src/unix/os390-syscalls.c

```diff
@@ -288,7 +290,8 @@ int epoll_wait(uv__os390_epoll* lst, struct epoll_event* events,
 
   reventcount = 0;
   nevents = 0;
-  for (int i = 0; 
+  msg_fd = pfds[lst->size - 1];
```

Comment:
Is it guaranteed that `lst->size > 0` at this point?

Reply:
@bnoordhuis good point.
```c
  _SET_FDS_MSGS(size, 1, lst->size - 1);
  pfds = lst->items;
  pollret = poll(pfds, size, timeout);
  if (pollret <= 0)
    return pollret;
```
This code, which is execute before the line you mentioned, makes an assumption that there is at least the message queue. Then it passes the items array to the poll() call, which would be NULL if the size was 0. That'd be a bug on its own. From my experience, passing NULL on z/OS might not lead to a crash. Let me try to break the new and the original function, see how it works when such list is passed.

Reply:
`lst->size` is unsigned so `_SET_FDS_MSGS(size, 1, lst->size - 1)` is equivalent to `_SET_FDS_MSGS(size, 1, UINT64_MAX)` when `lst->size == 0`.

I don't know what that macro does exactly but I bet that's not the intended behavior. :-)

Reply:
Yep, that would cause problems :) There is a single epoll_wait call site, which is never passed `lst` with size 0 (or we've been lucky). I can add an `if (lst->size == 0) return 0;` check at the top of epoll_wait in case it gets used some place else in the future. Though, that should also be carefully examined, because that call has some assertions after, like
``` c
nfds = epoll_wait(loop->ep, events, ARRAY_SIZE(events), timeout);
. . .
if (nfds == 0) {
  assert(timeout != -1);
. . .
}
```

Reply:
@bnoordhuis I've done some experiments with passing bad parms to epoll_wait. Those were `lst` with `size == 0`, `lst` with `items == NULL` and `lst` with both `size == 0` and `items == NULL`. Neither of those went past the poll() call. It'd always fail with either EINVAL or EFAULT, and eventually it would lead to an `abort()` call since the only acceptable errno to the (only) caller is EINTR. I'd say with the current caller's code, a situation when `size == 0` is passed will lead to termination one way or another but keeping the current logic leads to a result that closer resembles the Linux epoll behavior -  it fails with EINVAL if you create an epoll file descriptor with a non-positive size. In other words, checking the size and returning 0 if the size is 0 would be less "standard" behavior for epoll_wait.

Reply:
Okay. Can you add an `assert(lst->size > 0);` just in case then?

Reply:
^ This comment is still unaddressed.

Reply:
I've proposed a different set of asserts in [my other comment](https://github.com/libuv/libuv/pull/2013#discussion_r297526918) which should cover this case.